pages/login/LoginHandler.php

```diff
@@ -129,18 +129,22 @@ public function signIn(array $args, PKPRequest $request): void
             $request->redirectSSL();
         }
 
-        $error = null;
         $isCaptchaEnabled = Config::getVar('captcha', 'captcha_on_login') && Config::getVar('captcha', 'recaptcha');
         if ($isCaptchaEnabled) {
             $templateMgr->assign('recaptchaPublicKey', Config::getVar('captcha', 'recaptcha_public_key'));
             try {
                 FormValidatorReCaptcha::validateResponse($request->getUserVar('g-recaptcha-response'), $request->getRemoteAddr(), $request->getServerHost());
             } catch (Exception $exception) {
-                $error = 'common.captcha.error.missing-input-response';
+                $recaptchaError = 'common.captcha.error.missing-input-response';
             }
         }
 
-        $error = $this->_validateAltchasResponse($request, 'altcha_on_login');
+        $isAltchaEnabled = Config::getVar('captcha', 'altcha_on_login') && Config::getVar('captcha', 'altcha');
+        if ($isAltchaEnabled) {
+            $altchaError = $this->_validateAltchasResponse($request, 'altcha_on_login');
+        }
```

Comment:
Assuming `altcha_on_login` and `recaptcha_on_login` work properly together, then if we're not going to show both errors, the validation below can be skipped if the `$recaptchaError` has a value.